fix: (Platform) fix non-collapsible Dynamic Page header

KevinOkamoto opened the pull request:
#### Please provide a link to the associated issue.

#3794 

#### Please provide a brief summary of this pull request.
This PR:

- Fixes issue of clicking on title area collapses the Dynamic Page header when header attribute `collapsible=false`
- Adds Non-Collapsible header example to documentation
- Adds Non-Collapsible suite of unit tests


#### Please check whether the PR fulfills the following requirements

- [x] the commit message follows the guideline:
https://github.com/SAP/fundamental-ngx/blob/master/CONTRIBUTING.md
- [x] tests for the changes that have been done
- [x] all items on the PR Review Checklist are addressed :
https://github.com/SAP/fundamental-ngx/wiki/PR-Review-Checklist

Documentation checklist:
- [x] Documentation Examples

kavya-b commented on libs/platform/src/lib/components/dynamic-page/dynamic-page.component.html:
```diff
@@ -1,5 +1,5 @@
 <section class="fd-dynamic-page" [attr.aria-label]="ariaLabel">
-    <header class="fd-dynamic-page__header-fixed" #header>
+    <header class="fd-dynamic-page__header-fixed" [class]="{'fd-dynamic-page__header--not-collapsible': !headerCollapsible}" #header>
```
I don't see this class's styles getting applied in the example: 
![image](https://user-images.githubusercontent.com/53509521/98761517-9811d100-23fb-11eb-82f0-0abe8b951ea6.png)

KevinOkamoto replied:
This is because the style works on the title area not on the page header element itself:

<img width="1680" alt="Screen Shot 2020-11-13 at 7 50 21 AM" src="https://user-images.githubusercontent.com/48103491/99091695-1352be80-2585-11eb-941a-4cf18b92cfbf.png">

And it is only active on 'hover' or 'focus' of the title area.

InnaAtanasova commented:
1. You should be able to expand/collapse not only by clicking on the header but with the `Enter` key as well (for a11y reasons)
2. The pin button shows/hides depending on the state of the header:
<img width="2214" alt="Screen Shot 2020-11-12 at 2 21 27 PM" src="https://user-images.githubusercontent.com/39598672/98986424-e3f37180-24f2-11eb-96bf-f80894cb518f.png">
<img width="2236" alt="Screen Shot 2020-11-12 at 2 21 35 PM" src="https://user-images.githubusercontent.com/39598672/98986430-e6ee6200-24f2-11eb-92c8-5bdc7b6dc056.png">

KevinOkamoto commented:
@InnaAtanasova, I made the following fixes:

* added expand/collapse of header on keyboard enter when the title area is in focus
* refactored the "pinning" feature to follow UI5 behavior
  - pinning does not hide the collapse button
  - collapsing the header clicking the collapse button or title area overrides the pin and collapses the header
  - upon collapsing the header, if the header was "pinned" it becomes "unpinned"

kavya-b reviewed (approved):
Looks good, thanks Kevin!

Betrozov commented on libs/platform/src/lib/components/dynamic-page/dynamic-page.component.html:
```diff
@@ -1,7 +1,7 @@
 <section class="fd-dynamic-page" [attr.aria-label]="ariaLabel">
-    <header class="fd-dynamic-page__header-fixed" #header>
+    <header class="fd-dynamic-page__header-fixed" [class]="{'fd-dynamic-page__header--not-collapsible': !headerCollapsible}" #header>
```
`[class.fd-dynamic-page__header--not-collapsible]="!headerCollapsible"`

KevinOkamoto replied:
ok done

Betrozov commented on apps/docs/src/app/platform/component-docs/platform-dynamic-page/platform-dynamic-page-examples/platform-dynamic-page-non-collapsible-example.component.html:
```diff
@@ -0,0 +1,155 @@
+                    <!-- breadcrumb content -->
+                    <fd-breadcrumb>
+                        <fd-breadcrumb-item>
+                            <a fd-breadcrumb-link [attr.href]="'#'">Men</a>
```
why do you need here binding?
can we use like this `href="#"`

KevinOkamoto replied:
done

JKMarkowski reviewed (approved):
LGTM
